.github/actions/e2e_recover/action.yml

```diff
@@ -68,6 +68,26 @@ runs:
             echo "Did not recover all nodes yet, retrying in 5 seconds [$recovered/${{ inputs.controlNodesCount }}]"
             sleep 5
         done
-        kubectl wait --for=condition=Ready --timeout=10m --all nodes
+    - name: Wait for control plane to get back up
+      shell: bash
+      run: |
+        timeout=600
+        start_time=$(date +%s)
+        while true; do
+          output=$(kubectl wait --for=condition=Ready --timeout=10m --all nodes || true)
+          if echo "$output" | grep -q "condition met"; then
+              echo "$output"
+              exit 0
+          fi
+
+          current_time=$(date +%s)
+          if ((current_time - start_time > timeout)); then
+              echo "Waiting for control plane to get back up timed out after $timeout seconds."
+              exit 1
+          fi
+
+          echo "Cannot reach control plane: $output, retrying in 10 seconds"
```

Comment:
`$output` doesn't contain any value in this case, I assume because the error during `kubectl wait` is printed to stderr.

Maybe just remove `$output` from the message here?